emission/exportdata/export_data.py

```diff
@@ -0,0 +1,50 @@
+# Standard imports 
+import logging
+
+# Our imports
+import emission.storage.pipeline_queries as epq
+import emission.storage.timeseries.abstract_timeseries as esta
+import emission.storage.decorations.analysis_timeseries_queries as esda
+
+import gzip
+import json
+import bson.json_util as bju
+
+def set_export_data(user_id):
+    time_query = epq.get_time_range_for_export_data(user_id)        # Does this have .startTs and .endTs atributes? Should we use those?
+    try: 
+        edp = ExportDataPipeline()
+        edp.user_id = user_id
+        edp.run_export_data_pipeline(user_id, time_query)
+        if edp.last_trip_done is None:
+            logging.debug("After run, last_trip_done == None, must be early return")
+        epq.mark_export_data_done(user_id, edp.last_trip_done)
+    except:
+        logging.exception("Error while exporting, timestamp unchanged")
+        epq.mark_export_data_failed(user_id)
+
+class ExportDataPipeline:
+    def __init__(self):
+        self._last_trip_done = None
+
+    @property
+    def last_trip_done(self):
+        return self._last_trip_done
+
+    def run_export_data_pipeline(self, user_id, time_query):
+        ts = esta.TimeSeries.get_time_series(user_id)
+        
+        #This will essentially give us a list of the raw data in return. There are no restrictions on what is added, all keys are added so long as they fall in the time range.
+        raw_data_list = list(ts.find_entries(key_list=None, time_query=time_query))
+
```

Comment:
This is not strictly true. Because the query uses `tq.timeType = "data.end_ts"`, it will only return entries which have an `end_ts` field in their data, so "trip like" objects. It will not have any place objects and it will not have any raw location entries. If you look at the existing export script, it actually exports "trip like" objects, "place like" objects and "loc like" objects.

@kafitz will definitely want at least the "loc like" objects.

I would suggest creating a new directory within `emission` called `export` or something,
refactoring the `export_timeline` code into a module there, and calling it from both this code and from the standalone script.

Reply:
Common location, then call from there

Reply:
confirmed from the sample output that we are only getting "trip like" objects with the current query

```
$ zgrep "key" export_a6718c68-989e-4d22-96f7-ef16d4bcfd6c.gz  | sort | uniq
            "key": "analysis/cleaned_section",
            "key": "analysis/cleaned_trip",
            "key": "analysis/confirmed_trip",
            "key": "analysis/inferred_section",
            "key": "inference/prediction",
            "key": "segmentation/raw_section",
            "key": "segmentation/raw_trip",
```

Reply:
To follow up on the idea of refactoring, what I am thinking this will look like is: 

```
def run_export_data_pipeline(self, user_id, time_query):
        ts = esta.TimeSeries.get_time_series(user_id)

        raw_data_list = list(ts.find_entries(key_list=None, time_query=time_query))
```
I would take this original code, and move it into its own common location as we discussed. This is part of the class ExportDataPipeline. The remainder of the code would remain within the file. 

Questions: 
I am still a little confused if this is the code that we are looking to put in a common location (refactor). What do you mean by calling it from the standalone script? Am I thinking of this wrong?

Reply:
My idea for the refactored section would be to create a list that incorporates the trip data using "data.start_ts", the location data using "data.ts" and possibly the place data "data.enter_ts". This was inspired by the xtract_timeline_for_day_range_and_user.py file.

Reply:
@jruzekowicz 

> I would take this original code, and move it into its own common location as we discussed. This is part of the class ExportDataPipeline. The remainder of the code would remain within the file.

Not sure where there is original code called `run_export_data_pipeline`. Is this a new function you plan to write? If so, look at its arguments carefully.

Reply:
I understand that the time queries need to be changed to go along with the data that we want to export. I am confused however of why we need to refactor this, and what benefit is served there. Is this not something that could be better placed inside the ExportDataPipeline class? It seems like we would be creating two directories to execute parts of the same end goal. Can you clarify which parts we would want to put in a common location and what then calls that code?

Reply:
We need to refactor this so that we can reuse the export functionality from two locations without copy-pasting. Think through your software engineering classes on modularity and reuse and make a proposal.

Reply:
Okay, to clarify, which file(s) do you mean by extract_timeline... 
I currently see two files that start with that prefix, and they are located within bin/debug. Is this what I should be trying to make compatible with the refactored result?

Reply:
I mean the `extract_timeline_for_day_range_and_user.py` file that I linked to earlier and you have been using as a template. I was just lazy to type out the entire file name....

Reply:
Maybe I am missing something, but I thought we were going to change the export functionality for the pipeline state anyway? I though that we no longer wanted to export a .gz file. The extract_timeline_for_dat_range_and_user.py file still publishes a .gz file with the user_id as part of the name (very similar to my implementation and I can see how these could be used from a common location). However, do we want the new pipeline stage to export this data to a .gz file?

Reply:
1. Please see the discussion between @kafitz and I last night. Yes, this will export to a `.gz` file.
2. Even if it weren't, that doesn't preclude pulling out the common functionality. You would just define "common functionality" to be what is actually common between the two invocation locations.

Reply:
I just added a new commit to this pull request with an idea of how that could look. 
It is linked here: https://github.com/jruzekowicz/e-mission-server/commit/48389be2be1cee25fec8a43e8e6e1a07378ffc53 

The one major difference I see if the user cache usage as we talked before about not including that in the itinerum export data.